### PR TITLE
chore(e2e): increase timeouts, bump Detox, update macos runner

### PR DESCRIPTION
**Pull request by ioanabrooks**
#### Description of changes

This change:
* updates the timeout for app initialization to help address tests failures, both iOS and Android are timing out recently:
 https://github.com/aws-amplify/amplify-ui/actions/runs/5729414515/job/15531648865
 https://github.com/aws-amplify/amplify-ui/actions/runs/5729414515/job/15531649133

* refreshes yarn.lock file to update Detox version to latest.

* switches runner for ios tests to previous version (macos-12), the tests failures started happening with updated runners: https://github.com/actions/runner-images/issues/8027

* temporarily skip `Confirm new password page has correct translations` has been consistently failing in CI for both platforms, loses connection to Metro.

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/e2e/features/ui/components/authenticator/verify-user.feature [hunk 1]
```diff
@@ -19,7 +19,7 @@ Feature: Verify User
     Then "New Label" field does not have class "amplify-visually-hidden"
     Then I see placeholder "Enter your Confirmation Code:"
 
-  @react-nativee
+  @todo-react-native
```
Comment (ioanabrooks) [inline packages/e2e/features/ui/components/authenticator/verify-user.feature:22]: This tag was not meant as typo but tracked a follow-up work bc this test suite uses sign-in example app and mocks one API on web, but that is not possible with Detox (all apis or no apis get mocked) so this needs a separate example app to be able to run in RN.
Fix uninspired and confusing tag name.

Comment (reesscot) [reply packages/e2e/features/ui/components/authenticator/verify-user.feature:22]: Should we add a comment on why we are skipping it?

File: packages/e2e/.detoxrc.json [hunk 2]
```diff
@@ -42,7 +42,7 @@
     "simulator": {
       "type": "ios.simulator",
       "device": {
-        "type": "iPhone 14"
+        "type": "iPhone 13"
```
Comment (ioanabrooks) [inline packages/e2e/.detoxrc.json:45]: https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md#installed-simulators

File: .github/workflows/reusable-e2e.yml [hunk 3]
```diff
@@ -348,7 +348,7 @@ jobs:
   e2e-ios:
     needs: check-changed-packages
     if: ${{ inputs.skip-changed-packages-check == 'true' || needs.check-changed-packages.outputs.has-changed-packages == 'true' }}
-    runs-on: macos-13
+    runs-on: macos-12
```
Comment (reesscot) [inline .github/workflows/reusable-e2e.yml:351]: I know we probably need to do this given the issues with the runner, but MacOS 12 was slower, right?

**Review by thaddmt — APPROVED** [review]
LGTM

**Review by reesscot — APPROVED** [review]

**Pull request merged**